src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportStatus.java

```diff
@@ -39,33 +37,9 @@ public String getKey() {
     }
 
     @Override
-    public String getValue() {
-        if (StringUtils.isBlank(detail)) {
-            return event.getCode();
-        } else {
-            return detail + PLUS_SEPARATOR + event.getCode();
-        }
-    }
-
-    @Override
-    protected void validateStateful() throws EdifactValidationException {
-
-    }
-
-    @Override
-    public void preValidate() throws EdifactValidationException {
+    public void validate() throws EdifactValidationException {
         if (event.getCode().isBlank()) {
             throw new EdifactValidationException(getKey() + ": Status is required");
```

Comment:
If willing to go a little out of scope for tidy-up: we've tended to prefer just using `KEY` here directly instead of via `getKey()`
```suggestion
            throw new EdifactValidationException(KEY + ": Status is required");
```